src/python/pants/base/deprecated.py

```diff
@@ -118,7 +128,16 @@ def warn_or_error(removal_version, deprecated_entity_description, hint=None, sta
   :raises DeprecationApplicationError: if the removal_version parameter is invalid.
   :raises CodeRemovedError: if the current version is later than the version marked for removal.
   """
-  removal_semver = validate_removal_semver(removal_version)
+  removal_semver = validate_deprecation_semver(removal_version, 'removal version')
+  if deprecation_start_version:
+    deprecation_start_semver = validate_deprecation_semver(
+      deprecation_start_version, 'deprecation start version')
+    if deprecation_start_semver >= removal_semver:
+      raise InvalidSemanticVersionOrderingError(
+        'The deprecation start version {} must be greater than the end version {}.'
```

Comment:
less than

Reply:
This was fixed, and thoroughly tested via Eric's comment to mock out the PANTS_SEMVER.